js/src/main/scala/org/soundsofscala/instrument/SamplePlayer.scala

```diff
@@ -39,7 +54,18 @@ object SamplePlayer {
       _ <- IO(sourceNode.connect(gainNode))
       _ <- IO {
         sourceNode.buffer = buffer
-        sourceNode.playbackRate.value = playbackRate
+        sourceNode.loop = settings.loop
+        sourceNode.loopStart = settings.loopStart
+        sourceNode.loopEnd = settings.loopEnd
+        sourceNode.playbackRate.value = settings.playbackRate * playbackRate
+        if settings.reversed then {
```

Comment:
I have a feeling that if a reversed note (a call to `playSample`) is followed by a note that is not reversed, both will be played in reverse.

Reply:
Yes, but is that not desirable behaviour? If a sampler plays 1 sample then it should be the same for all the notes, including its manipulations. If one wanted to layer the reverse sounds over non-reversed one could create a new sampler/buffer that is not reversed.

Reply:
I see this as a design decision. 

The current design has `Settings` defined per note. So the `Settings` allow expressive control on a note-level. They correspond to, e.g., knob twiddling for a synth while the synth is playing. Under this model, if `Settings` allow a sample to be reversed it must be done per-note. I can play the sample normally one note, reversed the next, and back to normal on the third note.

If you want to change the semantics, so that reversing the sample is something done once when the instrument is created, and cannot be changed on a per-note basis, then either `Settings` needs to change or this is not something that can be controlled by `Settings`.